src/engine/guidance/lane_processing.cpp

```diff
@@ -27,9 +28,19 @@ std::vector<RouteStep> anticipateLaneChange(std::vector<RouteStep> steps,
 {
     // Lane anticipation works on contiguous ranges of quick steps that have lane information
     const auto is_quick_has_lanes = [&](const RouteStep &step) {
-        const auto is_quick = step.duration < min_duration_needed_for_lane_change;
         const auto has_lanes = step.intersections.front().lanes.lanes_in_turn > 0;
-        return has_lanes && is_quick;
+        if (!has_lanes)
+            return false;
+
+        // The more unused lanes to the left and right of the turn there are, the higher
+        // the chance the user is driving on one of those and has to cross lanes.
+        // Scale threshold for these cases to be adaptive to the situation's complexity.
+        const auto num_irrelevant = step.NumLanesToTheRight() + step.NumLanesToTheLeft();
```

Comment:
Can you explain your intention here?
From what I understand you are summing up all lanes left/right of the turn? Shouldn't you only care about the ones you actually have to cross? These should be either the ones to the left, or the ones to the right.
So what would happen if you are on the second lane from the left on a 16 lane highway and you want to turn left. Would you get 15 times the threshold, or only once?